server/backend/api/app/crud/crud_user_role.py

```diff
@@ -72,78 +74,118 @@ def create_user_role(
+                "Invalid user role assignment request, missing forest client number."
+            )
             raise_http_exception(
                 error_code=famConstants.ERROR_CODE_MISSING_KEY_ATTRIBUTE,
-                error_msg=error_msg
+                error_msg=error_msg,
             )
 
         api_instance_env = crud_utils.use_api_instance_by_app(fam_role.application)
-        validator = UserRoleValidator(request, api_instance_env)
-        if (not validator.forest_client_number_exists()):
-            error_msg = (
-                "Invalid role assignment request. " +
-                f"Forest Client Number {request.forest_client_number} does not exist.")
-            raise_http_exception(
-                error_code=famConstants.ERROR_CODE_INVALID_REQUEST_PARAMETER,
-                error_msg=error_msg
+        forest_client_integration_service = ForestClientService(api_instance_env)
+
+        for forest_client_number in request.forest_client_numbers:
+            # validate the forest client number
+            forest_client_validator_return = (
+                forest_client_integration_service.find_by_client_number(
+                    forest_client_number
+                )
             )
 
-        if (not validator.forest_client_active()):
-            error_msg = (
-                "Invalid role assignment request. Forest Client is not in Active status: " +
-                f"{validator.get_forest_client()[famConstants.FOREST_CLIENT_STATUS['KEY']]}")
-            raise_http_exception(
-                error_code=famConstants.ERROR_CODE_INVALID_REQUEST_PARAMETER,
-                error_msg=error_msg
+            if not forest_client_number_exists(forest_client_validator_return):
+                error_msg = (
+                    "Invalid role assignment request. "
+                    + f"Forest Client Number {forest_client_number} does not exist."
+                )
+                raise_http_exception(
+                    error_code=famConstants.ERROR_CODE_INVALID_REQUEST_PARAMETER,
+                    error_msg=error_msg,
+                )
+
+            if not forest_client_active(forest_client_validator_return):
+                error_msg = (
+                    f"Invalid role assignment request. Forest client number {forest_client_number} is not in active status:"
+                    + f"{get_forest_client_status(forest_client_validator_return)}"
+                )
+                raise_http_exception(
+                    error_code=famConstants.ERROR_CODE_INVALID_REQUEST_PARAMETER,
+                    error_msg=error_msg,
+                )
+
+            # Check if child role exists or add a new child role
+            child_role = find_or_create_forest_client_child_role(
+                db, forest_client_number, fam_role, requester
             )
+            # Role for associating with user
+            associate_role = child_role if require_child_role else fam_role
```

Comment:
This condition seems not necessary anymore, maybe can use `child_role` directly for `create_user_role_assignment`.